block/store.go

```diff
@@ -60,7 +60,7 @@ func (store *Store) Sync(ctx context.Context, prefix ds.Key) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
 	// Run the packer in a separate context to avoid canceling it prematurely.
-	store.packer.Run(context.Background())
+	store.packer.Start(context.Background())
```

Comment:
Note, you should be able to pass in the `ctx` as a parameter and then the packer itself should manage the goroutines so that it won't shutdown before all of the goroutines have finished.

Reply:
I don't really get what you say. `ctx` may be canceled very soon after `Start` starts the goroutine of the background job, and the method returns, which will undesirably cancel the background job too. The solution to this problem was to pass `context.Background()`. It's not ideal, but I didn't find a better place to trigger the start of the background job. Why should I pass `ctx`?